### ISSUE TITLE
Checklist: Single select option list uses buttons for single select widget

### ISSUE THREAD

**Issue by Blackbaud-MattGregg** [blackbaud/skyux1#796]
### Expected behavior

Single select option uses structural semantic markup for single select widget where group relationship and selected state available programmatically. 
### Actual behavior

Single option uses button elements instead of structural semantic markup. A set of radio buttons would indicate to screenreaders how many options are available in the group and have a checked state for what is selected. The buttons do not have a checked state which is programmatically available. These are read as a set of buttons that can be pushed and no action occurs.

**Comment by Blackbaud-PatrickOFriel** [comment]
The single select version does not actually use checkboxes. It uses buttons and updates a selected item field in angular that a developer can use in some way. "No action occurs" is not really accurate because whether clicked with a mouse or a keyboard you can see in the demo that the selected items field is updated.

**Comment by Blackbaud-MattGregg** [comment]
Yep. Sorry I wasn't clear. What I meant by no action occurs is from the perspective of someone who can't see the screen to understand what this is or what state it is in. To them, it's a list of buttons they can click and there is no change. The issue is this is a custom control using html elements for a different purpose than their default role. It uses a set of buttons (which don't communicate one selection is what needs be done or indicate a selected state). Radio buttons would be the correct semantic markup for a form input here. Communicating to someone this is a form input and have the selected state built in.

**Comment by Blackbaud-PatrickOFriel** [comment]
To me this is exactly the same as if there were a set of buttons in an application that had a given action occur on click. We do not make every set of buttons have radio roles, is that correct?

**Comment by Blackbaud-PatrickOFriel** [comment]
See the [selectfield single select](http://skyux.developer.blackbaud.com/components/selectfield/) for a case where this is actually used in applications.

**Comment by Blackbaud-MattGregg** [comment]
I've not made it to that component yet. The select component look pretty unique. I was evaluating this on its own which is how I've seen it. I wasn't evaluating it as part of the select field composite component. The select component looks like a different type of custom widget. 
For the checklist, I'm looking at it for that use. See opening a list (e.g. constits, gifts, etc.) in RE NXT or creating a new report wizard in FE NXT. Those are uses of the single select list on its own as a form control rather than part of a composite widget. A user is asked to choose one item in a list using this component on its own.

**Comment by Blackbaud-PatrickOFriel** [comment]
Gotcha, this is the first time I've seen it in that context. I understand why radio indication would be necessary now for cases like those

**Comment by Blackbaud-PatrickOFriel** [comment]
In [v1.7.3](https://github.com/blackbaud/skyux/releases/tag/v1.7.3)
